cassovary-core/src/main/scala/com/twitter/cassovary/graph/Traverser.scala

```diff
@@ -363,7 +363,7 @@ trait QueueBasedTraverser extends Traverser {
   /**
    * Performs action needed when visiting a node.
    */
-  protected def processNode(node : Node): Node = {
+  protected def processNode[V <: Node](node : V): V = {
```

Comment:
In next(), instead of 
```processNode(getExistingNodeById(graph, nextId))```
perhaps do this:
```
...
val node = getExistingNodeById(graph, nextId)
processNode(node)
node
```